src/components/carousel/carousel.js

```diff
@@ -0,0 +1,63 @@
+import mixin from '../../globals/js/misc/mixin';
+import createComponent from '../../globals/js/mixins/create-component';
+import initComponentBySearch from '../../globals/js/mixins/init-component-by-search';
+// import eventMatches from '../../globals/js/misc/event-matches';
+
+class Carousel extends mixin(createComponent, initComponentBySearch) {
+  /**
+   * Carousel.
+   * @extends CreateComponent
+   * @extends InitComponentBySearch
+   * @param {HTMLElement} element The element working as an carousel.
+   */
+  constructor(element, options) {
+    super(element, options);
+    this.filmstrip = this.element.querySelector(this.options.selectorFilmstrip);
+    this.carouselItem = this.element.querySelector(this.options.selectorCarouselItem);
+
+    this.element.addEventListener('click', evt => {
+      if (evt.target.matches(this.options.selectorScrollRight)) {
+        this.sideScroll('right');
+      }
+      if (evt.target.matches(this.options.selectorScrollLeft)) {
+        this.sideScroll('left');
+      }
+    });
+  }
+
+  sideScroll = direction => {
+    const filmstripWidth = this.filmstrip.getBoundingClientRect().width;
```

Comment:
We should avoid internal errors with missing `this.filmstrip`/`this.carouselItem`. Please see: https://github.com/carbon-design-system/carbon-components/pull/12